### PR TITLE
Feature/update dns record

### PR DESCRIPTION
**Pull request by alejandrojnm**
- Add option to updates DNS records

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -206,6 +206,22 @@ func (c *Client) GetDNSRecord(domainID, name string) (*DNSRecord, error) {
 	return nil, ErrDNSRecordNotFound
 }
 
+// UpdateDNSRecord updates the DNS record
+func (c *Client) UpdateDNSRecord(rc *DNSRecordConfig, d *DNSDomain, r *DNSRecord) (*DNSRecord, error) {
```
Comment (andyjeffries) [inline dns.go]: Why do you need to pass in `*DNSDomain` here? You can get the ID from `r.DNSDomainID` and not need to pass in an unnecessary object.

File: dns.go [hunk 2]
```diff
@@ -206,6 +206,22 @@ func (c *Client) GetDNSRecord(domainID, name string) (*DNSRecord, error) {
...
+	url := fmt.Sprintf("/v2/dns/%s/records/%s", d.ID, r.ID)
+	body, err := c.SendPutRequest(url, rc)
+	if err != nil {
+		return nil, err
+	}
+
+	var dnsDomain = &DNSRecord{}
```
Comment (andyjeffries) [inline dns.go]: This variable should be called `dnsRecord` rather than `dnsDomain` (as it's a Record object not a Domain)

File: dns_test.go [hunk 3]
```diff
@@ -204,6 +205,51 @@ func TestListDNSRecords(t *testing.T) {
...
+		  "name": "email",
+		  "value": "10.0.0.1",
+		  "type": "mx",
+		  "priority": 10,
+		  "ttl": 600
+		}`,
+	})
+	defer server.Close()
+	rc := &DNSRecordConfig{DNSDomainID: "edc5dacf-a2ad-4757-41ee-c12f06259c70", Name: "email"}
+	d := &DNSDomain{ID: "edc5dacf-a2ad-4757-41ee-c12f06259c70", AccountID: "1", Name: "example.com"}
```
Comment (andyjeffries) [inline dns_test.go]: With the change to the method signature, this object shouldn't be needed, but you will need to put `DNSDomainID` in the DNS Record object here.

**Comment by alejandrojnm** [comment]
I will make all change now, thx for the comments

**Comment by andyjeffries** [comment]
Merging this, but I'm going to change the parameter order and the configuration for updating them. I'll PM you a commit

**Pull request merged**